packages/trader/src/AppV2/Components/TradeParameters/PayoutPerPoint/payout-per-point-wheel.tsx

```diff
@@ -1,87 +1,118 @@
+        const new_values = { payout_per_point: String(value) };
+        const proposal_req = getProposalRequestObject({
+            new_values,
+            trade_store,
+            trade_type: Object.keys(trade_types)[0],
+        });
+        // Sending proposal without subscription to get a new barrier value
+        const { data: response } = useDtraderQuery<Parameters<TOnProposalResponse>[0]>(
+            [
+                'proposal',
+                ...Object.entries(new_values).flat().join('-'),
+                Object.keys(trade_types)[0],
+                JSON.stringify(proposal_req),
+            ],
+            proposal_req,
+            {
+                enabled: is_open,
             }
-            onWheelPickerScrollDebounced.cancel();
+        );
+
+        const onChange = (new_value: string | number) => {
+            // If a new value is equal to previous one, then we won't send API request
+            const is_equal = value === new_value;
+            is_api_response_received_ref.current = is_equal;
+            if (is_equal) return;
+
+            setValue(new_value);
         };
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
 
-    return (
-        <React.Fragment>
-            <ActionSheet.Content className='payout-per-point__wrapper' data-testid='dt_payout-per-point_wrapper'>
-                <div className='payout-per-point__wheel-picker'>
-                    <WheelPicker
-                        data={payout_per_point_list}
-                        selectedValue={selected_value_ref.current}
-                        setSelectedValue={(new_value: string | number) => {
-                            if (new_value === selected_value_ref.current) return;
-                            selected_value_ref.current = new_value;
-                            onWheelPickerScrollDebounced(new_value, onPayoutPerPointSelect);
-                        }}
-                    />
-                </div>
-                <div className='payout-per-point__barrier'>
-                    <Text color='quill-typography__color--subtle' size='sm'>
-                        <Localize i18n_default_text='Barrier' />
-                    </Text>
-                    <Text size='sm' as='div' className='payout-per-point__barrier__content'>
-                        {barrier ?? <Skeleton width={90} height={14} />}
-                    </Text>
-                </div>
-            </ActionSheet.Content>
-            <ActionSheet.Footer
-                alignment='vertical'
-                primaryAction={{
-                    content: <Localize i18n_default_text='Save' />,
-                    onAction: onSave,
-                }}
-            />
-        </React.Fragment>
-    );
-};
+        const onSave = () => {
+            // Prevent from saving if user clicks before BE validation
+            if (!is_api_response_received_ref.current) return;
+            onPayoutPerPointSelect(value);
```

Comment:
Only when user clicks on Save btn we are changing the values in trade store and sending them in main proposal (with subscription)